File: common-api/app/schemas/facebook_profile.py
from datetime import datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class FacebookProfileBase(BaseModel):
    facebook_id: str
    type: Literal["page", "user"]
    name: str
    profile_picture_url: str


class FacebookProfileCreate(FacebookProfileBase):
    pass


class FacebookProfileUpdate(BaseModel):
    facebook_id: str | None = None
    type: Literal["page", "user"] | None = None
    name: str | None = None
    profile_picture_url: str | None = None
    deleted_at: datetime | None = None


class FacebookProfileResponse(FacebookProfileBase):
    id: UUID
    created_at: datetime
    updated_at: datetime | None = None
    deleted_at: datetime | None = None
    model_config = ConfigDict(from_attributes=True)


class FacebookProfile(FacebookProfileResponse):
    pass

File: web/src/types/api/facebook-profile.ts
import type { FacebookProfileContact } from './facebook-profile-contact';

export type FacebookProfile = {
  facebook_id: string;
  type: 'page' | 'user';
  name: string;
  profile_picture_url: string;
  profile_contact?: FacebookProfileContact;
  username?: string;
  bio?: string;
};

export type FacebookProfileUpdate = Partial<FacebookProfile> & {
  deleted_at?: string;
};

export type FacebookProfileResponse = FacebookProfile & {
  id: string;
  created_at: string;
  updated_at?: string;
  deleted_at?: string;
};
